src/Stratis.SmartContracts.IntegrationTests/ContractExecutionFailureTests.cs

```diff
@@ -469,5 +471,61 @@ public void ContractTransaction_Create_IncorrectParameters()
             Assert.StartsWith(ContractInvocationErrors.MethodDoesNotExist, receipt.Error); // The error for constructor not found vs method does not exist could be different in future.
             Assert.Null(receipt.To);
         }
+
+
+        [Fact]
+        public void ContractTransaction_EmptyModule_Failure()
+        {
+            // Ensure fixture is funded.
+            this.node1.MineBlocks(1);
+
+            double amount = 25;
+            Money senderBalanceBefore = this.node1.WalletSpendableBalance;
+            uint256 currentHash = this.node1.GetLastBlock().GetHash();
+
+            // Create transaction with empty module
+            var module = ModuleDefinition.CreateModule("SmartContract", ModuleKind.Dll);
```

Comment:
Would be nice to see an `Assert(module.Types.Count == 1); Assert(module.Types[0].Name == "<Module>")` just to clarify what CreateModule is doing.

Reply:
Good idea, added